File: 프로그래머스/2/42586. 기능개발/기능개발.py
from collections import deque

def solution(progresses, speeds):
    answer = []
    progresses = deque(progresses)
    speeds = deque(speeds)
    
    while progresses:
        n = len(progresses)
        
        for i in range(n):
            progresses[i] += speeds[i]
            
        result = 0
        while progresses and progresses[0] >= 100:
            result += 1
            progresses.popleft()
            speeds.popleft()
        
        if result:
            answer.append(result)
    
    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    let answer = [];
    const n = progresses.length;
    
    let i = 0;
    while (i < n) {
        // 작업 진행하기
        progresses.forEach((item, index) => progresses[index] += speeds[index]);
        
        // 배포할 작업 계산하기
        let count = 0;
        while (progresses[i] >= 100) {
            count++;
            i++;
        }
        
        // 배포할 작업 있으면 반영하기
        if (count) answer.push(count);
    }
    
    return answer;
}